src/json/pull_parser.cr

```diff
@@ -126,6 +126,8 @@ class JSON::PullParser
       begin_array
     when .begin_object?
       begin_object
+    when .eof?
```

Comment:
Can we change all of these `when` to be `in`? I wonder if other cases are missing as well (probably not, but using `in` is better anyways)

Reply:
```
Missing members:
 - EndArray
 - EndObject
 - Colon
 - Comma
```

I think it makes sense to raise on these given it means the input is invalid JSON.  If we really wanted we could make this `in` and have these 4 members included in a single `in` that raises?

Reply:
I think that works! 👍